connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -686,21 +690,25 @@ protected Connector getConnector(String connType) {
         return tempConnectors.computeIfAbsent(connType, k -> plugins().newConnector(k));
     }
 
-    /**
-     * Retrieves ConnectorType for the corresponding connector class
-     * @param connClass class of the connector
-     */
-    public ConnectorType connectorTypeForClass(String connClass) {
-        return ConnectorType.from(getConnector(connClass).getClass());
-    }
-
     /**
      * Retrieves ConnectorType for the class specified in the connector config
-     * @param connConfig the connector config; may not be null
-     * @return the {@link ConnectorType} of the connector
+     * @param connConfig the connector config, may be null
+     * @return the {@link ConnectorType} of the connector, or {@link ConnectorType#UNKNOWN} if an error occurs
```

Comment:
Nit:
```suggestion
     * @return the {@link ConnectorType} of the connector, or {@link ConnectorType#UNKNOWN} if an error occurs or the type cannot be determined
```
(Since we gracefully handle null input by returning `UNKNOWN`, we shouldn't imply that it's only returned if an error occurs.)